src/memote/experimental/essentiality.py

```diff
@@ -92,7 +93,9 @@ def evaluate(self, model):
             essen = single_gene_deletion(
                 model, gene_list=self.data["gene"], processes=1
             )
-        essen["gene"] = [list(g)[0] for g in essen.index]
+        essen["gene"] = [
+            list(g)[0] if isinstance(g, Iterable) else g for g in essen["ids"]
```

Comment:
Aren't elements of `essen["ids"]` always sets?

Reply:
Right, that filter was just clarifying the error, but it is superfluous now.